src/plugins/vue/VuePlugin.ts

```diff
@@ -0,0 +1,266 @@
+import { File } from "../../core/File";
+import { WorkFlowContext, Plugin } from "../../core/WorkflowContext";
+import { CSSPluginClass } from "../stylesheet/CSSplugin";
+import { Concat, hashString } from "../../Utils";
+import { VueBlockFile } from './VueBlockFile';
+import { VueTemplateFile } from './VueTemplateFile';
+import { VueStyleFile } from './VueStyleFile';
+import { VueScriptFile } from './VueScriptFile';
+import * as path from "path";
+import * as fs from "fs";
+import { each } from "realm-utils";
+const vueCompiler = require("vue-template-compiler");
+const DEFAULT_OPTIONS: IVueComponentPluginOptions = {
+  script: [],
+  template: [],
+  style: []
+};
+
+export interface IVueComponentPluginOptions {
+  script?: Plugin[],
+  template?: Plugin[],
+  style?: Plugin[]
+}
+
+export class VueComponentClass implements Plugin {
+  public dependencies: ["fusebox-hot-reload"];
+  public test: RegExp = /\.vue$/
+  public options: IVueComponentPluginOptions;
+
+  constructor(options: IVueComponentPluginOptions) {
+    this.options = Object.assign({}, DEFAULT_OPTIONS, options);
+    this.options.script = Array.isArray(this.options.script) ? this.options.script : [this.options.script];
+    this.options.template = Array.isArray(this.options.template) ? this.options.template : [this.options.template];
+    this.options.style = Array.isArray(this.options.style) ? this.options.style : [this.options.style];
+  }
+
+  public init(context: WorkFlowContext) {
+      context.allowExtension(".vue");
+  }
+
+  private getDefaultExtension(block: any) {
+    switch (block.type) {
+      case 'template':
+        return 'html';
+      case 'script':
+        return 'js';
+      case 'style':
+        return 'css';
+    }
+  }
+
+  private createVirtualFile(file: File, block: any, scopeId: string, pluginChain: Plugin[]): VueBlockFile {
+    let extension = block.lang || this.getDefaultExtension(block);
+    let src = `./${block.type}.${extension}`;
+
+    if (block.src) {
+      extension = path.extname(block.src);
+      src = extension ? block.src : `${block.src}.${block.lang || this.getDefaultExtension(block)}`;
+    }
+
+    const fileInfo = file.collection.pm.resolve(src, file.info.absDir);
+
+    switch (block.type) {
+      case 'script':
+        return new VueScriptFile(file.context, fileInfo, block, scopeId, pluginChain);
+      case 'template':
+        return new VueTemplateFile(file.context, fileInfo, block, scopeId, pluginChain);
+      case 'style':
+        return new VueStyleFile(file.context, fileInfo, block, scopeId, pluginChain);
+    }
+  }
+
+  private addToCacheObject(cacheItem: any, path: string, contents: string, sourceMap: any) {
+    cacheItem[path] = {
+      contents,
+      sourceMap
+    };
+  }
+
+  public bundleEnd(context: WorkFlowContext) {
+    if (context.useCache) {
+      context.source.addContent(`
+        var process = FuseBox.import('process');
```

Comment:
Let's make sure process is included for sure. Put it into file dependencies.
And I would suggest wrapping it
```
if( process.env.NODE_ENV !== "production) {
 // that code
}
```

Reply:
Updated.